File: Frontend/src/app/models/seguimiento.dto.ts
export interface EtapaSeguimientoDTO {
  nombre: string;
  estadoVisual: string; // COMPLETADO, EN_PROCESO, PENDIENTE, RECHAZADO
  fecha: string | null;
  descripcion: string;
}

export interface SeguimientoDTO {
  solicitudId: number;
  tituloProyecto: string;
  estadoActual: string;
  porcentajeProgreso: number;
  etapas: EtapaSeguimientoDTO[];
}

File: backend/src/main/java/ec/edu/uteq/presustentaciones/dto/SeguimientoDTO.java
package ec.edu.uteq.presustentaciones.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class SeguimientoDTO {
    private Long solicitudId;
    private String tituloProyecto;
    private String estadoActual; // Estado general de la solicitud
    private int porcentajeProgreso;
    private List<EtapaSeguimientoDTO> etapas;
}
